src/main/scala/edu/ie3/simona/model/thermal/ThermalHouse.scala

```diff
@@ -107,24 +107,28 @@ final case class ThermalHouse(
     /* Determine, which temperature boundary triggers a needed energy to reach the temperature constraints */
     val temperatureToTriggerRequiredEnergy =
       if (
-        innerTemperature <= state.innerTemperature &&
+        currentInnerTemp <= state.innerTemperature &&
         state.qDot <= Kilowatts(0d)
-      )
+      ) {
+        // temperature has been decreasing and heat source has been turned off
+        // => we have reached target temp before and are now targeting lower temp
         lowerBoundaryTemperature
-      else targetTemperature
+      } else targetTemperature
```

Comment:
As discussed we should handle this within a new issue
- [ ] checking if this method catch all cases
- [ ] make easier understandble what happens here